tests/unit/test_solvers/test_idaklu_solver.py

```diff
@@ -1162,3 +1162,23 @@ def test_python_idaklu_deprecation_errors(self):
                     match="Unsupported evaluation engine for convert_to_format=jax",
                 ):
                     _ = solver.solve(model, t_eval)
+
+    def test_extrapolation_events_with_output_variables(self):
+        # Make sure the extrapolation checks work with output variables
+        model = pybamm.BaseModel()
+        v = pybamm.Variable("v")
+        c = pybamm.Variable("c")
+        model.variables = {"v": v, "c": c}
+        model.rhs = {v: -1, c: 0}
+        model.initial_conditions = {v: 1, c: 2}
+        model.events.append(
+            pybamm.Event(
+                "Ignored event",
+                v + 10,
+                pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+            )
+        )
+        solver = pybamm.IDAKLUSolver(output_variables=["c"])
+        solver.set_up(model)
+
+        solver.solve(model, t_eval=[0, 1])
```

Comment:
Can we add a check to make sure it's returning the appropriate `t` and `y` vals?

Reply:
I've switched the event definition instead to check it's going through `check_extrapolation` and calculating correctly by raising a SolverWarning